### ISSUE TITLE
[BUG] Bar graph crashes Julia kernel with the wrong syntax

### ISSUE THREAD

**Issue by biona001** [JuliaPlots/Plots.jl#4203]
## Details
I struggled to made the following bar graph
```julia
using Plots
n = 100
x = rand(n)
colors = [x[i] < 0.5 ? :red : :blue for i in 1:n]
labels = ["" for i in 1:n]
labels[findfirst(x -> x == :blue, colors)] = "None-null"
labels[findfirst(x -> x == :red, colors)] = "Null"
bar(
    permutedims(collect(1:n)),
    permutedims(rand(n)),
    color = permutedims(colors),
    labels = permutedims(labels)
)
```
![download](https://user-images.githubusercontent.com/16760873/168492221-3de2be7b-0259-438d-abf2-d1ee41686f4e.png)

Basically, **the following syntax will crash Julia** (although the code is essentially the same as the working version above but not transposed)
```julia
# dies
bar(
    x,
    color = colors,
    labels = labels
)

# dies
bar(
    1:n,
    x,
    color = colors,
    labels = labels
)
```

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |   x  |     |
pyplot       |     |     |x
plotlyjs     |     |     |x
pgfplotsx    |     |   x  |
unicodeplots |     |     |x
inspectdr    |     |     |x
gaston       |     |     |x

### Versions

Plots.jl version: `[91a5bcdd] Plots v1.29.0`
Backend  version (`]st -m <backend(s)>`): too long
Output of `versioninfo()`:
```
Julia Version 1.7.1
Commit ac5cc99908 (2021-12-22 19:35 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin19.5.0)
  CPU: Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, skylake)
```

**Comment by BeastyBlacksmith** [comment]
I am not saying
```julia
bar(
    1:n,
    x,
    color = colors,
    labels = labels
)
```
should result in the picture above, since one has to provide row vectors for that. But you shouldn't get segmentation faults, if you do that. I am assuming it's because of the long label and thus a duplicate of https://github.com/JuliaPlots/Plots.jl/issues/4198.
Could you add a check in GR.jl that the label fits the buffer and throw an error otherwise?

**Comment by jheinen** [comment]
Ahh, I understand. When I pass a subset (`labels = labels[1:10]`), it doesn't crash. Will try fix this within GR.jl ...

**Comment by jheinen** [comment]
The problem is fixed in GR 0.64.3.
